invenio_rdm_records/resources/serializers/schemas.py

```diff
@@ -12,6 +12,19 @@
 class BaseSchema(Schema):
     """BaseSchema for serializers in JSON."""
 
+    def get_doi(self, obj):
+        """Get DOI."""
+        item = [
```

Comment:
I think the idea is the opposite:

1. Try `pids`. This field has more checks and constraints so it is more "trust worthy" and should be taken into account first (right @kpsherva?)
2. Then default to identifiers

In addition, I think early return could be use to simplify the logic of this function. Creating lists to then return `iten[0]` is a bit counter intuitive. In addtion, it means that we would be looping through all identifiers when in reality we only need the first one (i.e. we could save processing time). For example, as follows (pseudocode):

```python
if "doi" in ...:
    return obj[...]

for identifier in obj[...]:
    if identifier["scheme"] == "doi":
        return ...

return missing
```

Since we are using returns we can also avoid using nested if-else.